api/v1beta1/grafana_types.go

```diff
@@ -119,6 +119,9 @@ type GrafanaClient struct {
 	// TLS Configuration used to talk with the grafana instance.
 	// +optional
 	TLS *TLSConfig `json:"tls,omitempty"`
+	// Optional map of header key and value.
+	// +optional
+	Headers *map[string]string `json:"headers,omitempty"`
```

Comment:
```suggestion
	Headers map[string]string `json:"headers,omitempty"`
```
I think there's no need to use a pointer with a map, we don't do it anywhere in our codebase.